### ISSUE TITLE
Boxplot: Set Whisker options

### ISSUE THREAD

**Issue by jdittrich** [plotly/plotly.js#277]
**Problem:** [Boxplot Whiskers can map to different data](https://en.wikipedia.org/wiki/Box_plot#Types_of_boxplots) – sometimes this is +- 1.5 times the interquartil range, sometime the 5th and 95th percentile etc. In the [box documentation](https://plot.ly/javascript/reference/#box), it is state that plotly uses the +-1.5 IQR as a standard. It is also stated that the _boxpoints_ option allows other options, but seemingly _boxpoints_ sets the  displaying sof catterpoint and not of whiskers.

**Proposed Solution:** An option for setting what the whiskers mean. Candidates for this option would: 
- n \* IQR (for a +-1.5\* the IQR, like the current default)
- nth percentiles (for something like a 5%/95% position)
- n*sd (for a +- 1 standard deviation position)

Or, if that option(s) already exists, a hint in the docuementation on how to set them.

**Comment by etpinard** [comment]
Thanks for the very clear description :beers: 

You are correct. There is no way to override the box trace whisker metric at the moment. 

Adding whiskers option is a great suggestion!

Unfortunately, because of other obligations, no plotly.js team member will dedicate time to this enhancement until at least April 1st. 

That said, this issue could be a great way for someone from the community to make a PR. All the action will take place in [`src/traces/box/`](https://github.com/plotly/plotly.js/tree/master/src/traces/box) starting by adding a trace attribute in [`src/traces/box/attributes.js`](https://github.com/plotly/plotly.js/tree/master/src/traces/box/attributes.js) then coercing it in the default step [`src/traces/box/defaults.js`](https://github.com/plotly/plotly.js/blob/master/src/traces/box/defaults.js) and finally generalizing the calc step in [`src/traces/box/calc.js`](https://github.com/plotly/plotly.js/blob/7b0036e2998028435be9e0608416eca0eb67fe1e/src/traces/box/calc.js#L108-L135).

**Comment by jdittrich** [comment]
:beers: cheers! 
Thanks for pointing out the relevant files. I see what I can do but can't promise that I'm quicker than April the 1st :-)

**Comment by Alanocallaghan** [comment]
@etpinard @jdittrich is this still unresolved?

**Comment by jdittrich** [comment]
@Alanocallaghan sadly not, I was unable to tackle it.

**Comment by alexcjohnson** [comment]
see also #1953 - one of the options should be max/min (which I guess would count as 0/100th percentile?) - and then rather than the current hard-coded dependence on `boxpoints`, `boxpoints` should just affect the default value of this setting.

**Comment by IanniF** [comment]
Hi, @etpinard @jdittrich is this still unresolved?

**Comment by FelixZY** [comment]
Any updates on this in 2019? Having the whiskers mean one thing in some graphs (where the boxpoints are visible) and another in others (where the boxpoints are not visible) is somewhat confusing.

**Comment by diegoscarabelli** [comment]
Agreed. Whiskers should correspond to editable percentile. That's how matplotlib does it.

**Comment by FilipVel** [comment]
Any update on this issue?

**Comment by ahmedhosny** [comment]
Seems like an old issue but keeps on coming back. I was a bit thrown off when I had `Boxpoints=False` but suddenly my whiskers extended to cover all the outliers and the entire plot changed.

![Screenshot from 2020-07-13 20-14-07](https://user-images.githubusercontent.com/8398334/87366295-586ad380-c546-11ea-8b95-ee2080f5f14b.png)

**Comment by jackparmer** [comment]
This issue has been tagged with `NEEDS SPON$OR`

A [community PR](https://github.com/plotly/plotly.js/pulls?q=is%3Apr+is%3Aopen+label%3A%22type%3A+community%22) for this feature would certainly be welcome, but our experience is deeper features like this are difficult to complete without the Plotly maintainers leading the effort.

**Sponsorship range: $5k-$10k**

What Sponsorship includes:
- Completion of this feature to the Sponsor's satisfaction, in a manner coherent with the rest of the Plotly.js library and API
- Tests for this feature
- Long-term support (continued support of this feature in the latest version of Plotly.js)
- Documentation at [plotly.com/javascript](https://plotly.com/javascript/)
- Possibility of integrating this feature with Plotly Graphing Libraries (Python, R, F#, Julia, MATLAB, etc)
- Possibility of integrating this feature with Dash
- Feature announcement on [community.plotly.com](https://community.plotly.com/tag/announcements) with shout out to Sponsor (or can remain anonymous)
- Gratification of advancing the world's most downloaded, interactive scientific graphing libraries (>50M downloads across supported languages)

Please include the link to this issue when [contacting us](https://plotly.com/get-pricing/) to discuss.

**Comment by gvwilson** [comment]
Hi - this issue has been sitting for a while, so as part of our effort to tidy up our public repositories I'm going to close it. If it's still a concern, we'd be grateful if you could open a new issue (with a short reproducible example if appropriate) so that we can add it to our stack. Cheers - @gvwilson

**Comment by rabyj** [comment]
Is there any open issue still tracking this feature? This has been bugging me for years, working from the Python side. 

For reference, there is a workaround to have the normal IQR whiskers while not having boxpoints shown:
`marker: {opacity: 0}`

**Comment by gvwilson** [comment]
there isn't, but I'll reopen this one - we're unlikely to get to it any time soon, but we'd be happy to prioritize review of a community PR. thanks - @gvwilson
